sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -3550,6 +3551,10 @@ object ArrayBinaryLike {
   def throwUnionLengthOverflowException(length: Int): Unit = {
     throw QueryExecutionErrors.unionArrayWithElementsExceedLimitError(length)
   }
+
+  def isNaN(value: Any): Boolean = {
+    Double.NaN.equals(value) || Float.NaN.equals(value)
```

Comment:
Seems `Double.NaN.equals(value)` can't work with `Scala-2.13` together, we need use `java.lang.Double.isNaN()` instead.

Reply:
Ok, done